pattern/constraint/thing/ValueConstraint.java

```diff
@@ -259,16 +263,10 @@ public Double asDouble() {
             }
 
             @Override
-            public boolean isConsistentWith(ValueConstraint.Constant<?> other) {
+            public <TYPE> boolean isConsistentWith(ValueConstraint.Constant<TYPE> other) {
                 if (other.predicate() != EQ) return true;
-                if (!valueEncoding.comparableTo(other.valueEncoding)) return false;
-                if (other.isDouble()) {
-                    return Predicate.Value.Numerical.of(predicate.asEquality(), PredicateArgument.Value.DOUBLE)
-                            .apply(other.asDouble().value(), value.doubleValue());
-                } else {
-                    return Predicate.Value.Numerical.of(predicate.asEquality(), PredicateArgument.Value.LONG)
-                            .apply(other.asLong().value, value);
-                }
```

Comment:
we eliminated the casting and checking here! Now we can actually just redirect into the `Predicate` to do all the required operations, which in turn uses `Encoding.ValueType.compare()`